ipn/ipnlocal/peerapi.go

```diff
@@ -341,11 +341,71 @@ func (s *peerAPIServer) DeleteFile(baseName string) error {
 // accidentally logging actual filenames anywhere.
 const redacted = "redacted"
 
+type redactedErr struct {
+	msg   string
+	inner error
+}
+
+func (re *redactedErr) Error() string {
+	return re.msg
+}
+
+func (re *redactedErr) Unwrap() error {
+	return re.inner
+}
+
 func redactErr(err error) error {
-	if pe, ok := err.(*os.PathError); ok {
+	var redactStrings []string
```

Comment:
golang/go#53435 recently added the concept of a multi-error. Effectively, errors in Go are no longer a chain, but rather a tree. The use of `errors.As` unfortunately will not catch all possible cases.

In that proposal, there was some discussion about whether you can iterate or split apart a multi-error, but it was deemed both esoteric and/or unclear what the use-case would be.

This PR presents a use-case for such a feature.

Reply:
I sent out #6643, which could augment this PR.